torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1378,13 +1389,22 @@ Tensor binary_cross_entropy_target_backward(
 Tensor binary_cross_entropy_with_logits_target_backward(const Tensor& grad_output, const Tensor& self, const Tensor& target, const c10::optional<Tensor>& weight, const c10::optional<Tensor>& pos_weight, int64_t reduction) {
   Tensor grad_target;
   if (isDefined(pos_weight)) {
-    grad_target = (1. - self.sigmoid()).log_().sub_(pos_weight->mul(self.sigmoid().log_())).mul_(grad_output);
+    if (!areAnyTensorSubclassLike({*pos_weight, grad_output})) {
+      // fast-path
+      grad_target = (1. - self.sigmoid()).log_().sub_(pos_weight->mul(self.sigmoid().log_())).mul_(grad_output);
+    } else {
+      grad_target = (1. - self.sigmoid()).log_().sub(pos_weight->mul(self.sigmoid().log_())).mul(grad_output);
```

Comment:
the mul with self.sigmoid() could be inplace no?

Reply:
Ho that op is applied to pos_weight directly. Sorry with the line break I though it was an intermediary. We don't want to modify pos_weight inplace so we shouldn't indeed.